File: app/src/main/java/dev/blackcat/minauta/AccountActivity.java
package dev.blackcat.minauta;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import dev.blackcat.minauta.data.Account;
import dev.blackcat.minauta.data.store.PreferencesStore;

public class AccountActivity extends MyAppCompatActivity
{

    EditText usernameEdit;
    EditText passwordEdit;
    Button acceptButton;

    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_account);

        this.initComponents();
    }

    private void initComponents()
    {
        final PreferencesStore prefsData = new PreferencesStore(this);

        Account account = prefsData.getAccount();
        usernameEdit = this.getViewById(R.id.usernameEdit);
        passwordEdit = this.getViewById(R.id.passwordEdit);
        if (account != null)
        {
            usernameEdit.setText(account.getUsername());
            passwordEdit.setText(account.getPassword());
        }

        acceptButton = this.getViewById(R.id.acceptButton);
        acceptButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view)
            {
                String username = usernameEdit.getText().toString();
                String password = passwordEdit.getText().toString();

                if (!username.contains("@"))
                    username += "@nauta.com.cu";
                prefsData.setAccount(username, password);

                finish();
            }
        });
    }

}

File: app/src/main/java/dev/blackcat/minauta/ui/account/AccountActivity.kt
package dev.blackcat.minauta.ui.account

import android.os.Bundle
import android.widget.EditText
import android.widget.Spinner
import androidx.appcompat.app.AlertDialog
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import com.google.android.material.floatingactionbutton.FloatingActionButton
import dev.blackcat.minauta.R
import dev.blackcat.minauta.data.AccountType
import dev.blackcat.minauta.ui.MyAppCompatActivity

class AccountActivity : MyAppCompatActivity() {

    lateinit var viewModel: AccountViewModel

    lateinit var accountsView: AccountRecyclerView
    lateinit var addAccountButton: FloatingActionButton

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_account)

        val dialog = AccountDialog(this)

        viewModel = ViewModelProvider.AndroidViewModelFactory(application).create(AccountViewModel::class.java)
        viewModel.accountsLiveData.observe(this, Observer { accounts ->
            accountsView.setStoredAccounts(accounts)
        })

        accountsView = findViewById(R.id.accountsRecyclerView)
        accountsView.initComponent()
        accountsView.setStoredAccounts(viewModel.getStoredAccounts())
        accountsView.setOnEditAccount { position ->
            val account = viewModel.getStoredAccount(position)
            account?.let { account ->
                dialog.editAccount(account) { account ->
                    viewModel.updateStoredAccount(position, account)
                }
            }
        }
        accountsView.setOnRemoveAccount { position ->
            dialog.removeAccount {
                viewModel.removeStoredAccount(position)
            }
        }
        accountsView.setOnSelectAccount { position ->
            viewModel.selectStoredAccount(position)
        }

        addAccountButton = findViewById(R.id.addAccountButton)
        addAccountButton.setOnClickListener { v ->
            dialog.createAccount { account ->
                viewModel.addStoredAccount(account)
            }
        }
    }

}